src/gpdma/config/transform.rs

```diff
@@ -0,0 +1,319 @@
+}
+
+impl PaddingAlignment for DataTransformBuilder<u8, u32> {
+    /// Pad out the upper 24 bits of the 32-bit destination word with zeroes (default)
+    fn right_align_zero_pad(mut self) -> Self {
+        self.transform.padding_alignment = PaddingAlignmentMode::ZeroPadded;
+        self
+    }
+
+    /// Sign extend the upper 24 bits of the 32-bit destination word
+    fn right_align_sign_extend(mut self) -> Self {
+        self.transform.padding_alignment = PaddingAlignmentMode::SignExtended;
+        self
+    }
+
+    /// Pack subsequent 8-bit words into the 32-bit destination words
+    /// ie: B3,B2,B1,B0 -> B3B2B1B0 (see RM0492, Table 92)
+    fn pack(mut self) -> Self {
+        self.transform.padding_alignment = PaddingAlignmentMode::Packed;
+        self
+    }
+}
+
+impl PaddingAlignment for DataTransformBuilder<u8, u16> {
+    /// Pad out the upper 8 bits of the 16-bit destination word with zeroes (default)
+    fn right_align_zero_pad(mut self) -> Self {
+        self.transform.padding_alignment = PaddingAlignmentMode::ZeroPadded;
+        self
+    }
+
+    /// Sign extend the upper 8 bits of the 32-bit destination word
+    fn right_align_sign_extend(mut self) -> Self {
+        self.transform.padding_alignment = PaddingAlignmentMode::SignExtended;
+        self
+    }
+
+    /// Pack subsequent 8-bit words into the 16-bit destination words
+    /// ie: B1,B0 -> B1B0 (see RM0492, Table 92)
+    fn pack(mut self) -> Self {
+        self.transform.padding_alignment = PaddingAlignmentMode::Packed;
+        self
+    }
+}
+
+impl TruncationAlignment for DataTransformBuilder<u32, u16> {
+    /// Keep the least significant 16-bits and truncate the rest (default)
+    ///
+    /// ie: B7B6B5B4,B3B2B1B0 -> B5B4,B1B0 (see RM0492, Table 92)
+    fn right_align_left_truncate(mut self) -> Self {
+        self.transform.padding_alignment =
+            PaddingAlignmentMode::RightAlignedLeftTruncated;
+        self
+    }
+
+    /// Keep the most significant 16-bits and truncate the rest
+    ///
+    /// ie: B7B6B5B4,B3B2B1B0 -> B7B6,B3B2 (see RM0492, Table 92)
+    fn left_align_right_truncate(mut self) -> Self {
+        self.transform.padding_alignment =
+            PaddingAlignmentMode::LeftAlignedRightTruncated;
+        self
+    }
+
+    /// Unpack each 32-bit word into separate 16-bit half-words.
+    /// Note that the destination buffer must have sufficient room for n*2 16-bit values where n is
+    /// the number of 32-bit words in the source buffer.
+    ///
+    /// ie: B7B6B5B4,B3B2B1B0 -> B7B6,B5B4,B3B2,B1B0 (see RM0492, Table 92)
+    fn unpack(mut self) -> Self {
+        self.transform.padding_alignment = PaddingAlignmentMode::Unpacked;
+        self
+    }
+}
+
+impl TruncationAlignment for DataTransformBuilder<u32, u8> {
+    /// Keep the least significant 8-bits and truncate the rest (default)
+    ///
+    /// ie: B7B6B5B4,B3B2B1B0 -> B4,B0
+    fn right_align_left_truncate(mut self) -> DataTransformBuilder<u32, u8> {
```

Comment:
Nit, the other impl blocks has `Self` here
```suggestion
    fn right_align_left_truncate(mut self) -> Self {
```